src/fsharp/FSharp.Core/string.fs

```diff
@@ -31,12 +31,15 @@ namespace Microsoft.FSharp.Core
 
         [<CompiledName("Map")>]
         let map (mapping: char -> char) (str:string) =
-            if String.IsNullOrEmpty str then
-                String.Empty
+            if String.IsNullOrEmpty str then String.Empty
             else
-                let res = StringBuilder str.Length
-                str |> iter (fun c -> res.Append(mapping c) |> ignore)
-                res.ToString()
+                let result = str.ToCharArray()
+                let mutable i = 0
+                for c in result do
+                    result.[i] <- mapping c
+                    i <- i + 1
+
+                String result
```

Comment:
```suggestion
                string result
```
I think this is what we're prefer. Is the `string` casing operator available at this point?

Reply:
Can't we explicitly call ToString()? That would make it clear that we definitely want the built in function and not something that F# might override in the future

Reply:
I think .ToString() is a good suggestion too.

Reply:
@cartermp I wanted that specific overload of the `String` constructor. I don't know if calling `ToString` on the array will have the same effect and call into the same methods internally, but I'll check. `string` adds overhead, that's probably not a good idea, but I'll run some timings.

Reply:
@cartermp, using `string` vs the constructor `String` won't work, neither will `ToString()`, see:

```f#
> string [|'a';'b'|];;
val it : string = "System.Char[]"

> [|'a';'b'|].ToString();;
val it : string = "System.Char[]"

> System.String [|'a';'b'|];;
val it : System.String = "ab"
```

We really need to call the `String: char[] -> string` constructor here, as 1) I don't have a good alternative and 2) it uses a highly optimized loop internally in native code.

I can make it explicit by using `System.String`, or add a comment so that nobody at some point will consider it is a typo.

Reply:
You could clarify by adding `new` and parens

Reply:
@NinoFloris, yeah, that'll work but `new` without `use` nowadays raises a warning, no? Anyway, anyone changing it to `string` will get tests failing, I suppose.

Reply:
@NinoFloris: done in https://github.com/dotnet/fsharp/pull/9470/commits/e2cde1b2fd517a72f0a77efbe7846827f2637b3c.